### DISCUSSION TITLE
python module get core dumped when trying to call a function in pure C shared library at runtime

### DISCUSSION THREAD

**Discussion by jundaf2** [pybind/pybind11#4636] [Q&A]
Hi!

(1) CMake successfully compiles the code (a mixture of C++ and C) and generates the python module, say`foo.cpython-310-x86_64-linux-gnu.so`. 
(2) Python script didn't report any linkage or python version error when importing the module `import foo`.
(3) Assume the `foo` module also depends on a pure C shared library `A.so` and `A.so` contains the definition of a method called `Run`. When the C++ source code of module `foo` (decorated by `extern "C"` for C linkage) trying to call the  `Run` routine in `A.so`, the program gets `Segmentation fault (core dumped)`.
(4) BTW, the same program also compiles and generates an executable in addition to the python module. The executable can successfully run without this problem with the same input.

Do you have any idea what's the problem and how I can avoid such puzzling issue?

Thanks

**Comment by AlexanderSerov** [comment]
Did you try attaching with debugger? See how to debug "Segmentation faults"

**Reply by K-Wu** [reply]
Hello Alexander,

Thank you for your reply. Junda and I have been investigating into this these days and figured out the seg fault is caused by the function name conflict in another library with re_comp in glibc. We submitted an issue in the corresponding repository https://github.com/katef/libfsm/issues/425

Thank you again.

Best Regards,
Kun
